src/dashboards/components/DashboardContainer.tsx

```diff
@@ -26,12 +26,71 @@ const {Active} = AutoRefreshStatus
 type ReduxProps = ConnectedProps<typeof connector>
 type Props = ReduxProps
 
+let timer
 const DashboardContainer: FC<Props> = ({autoRefresh, dashboard}) => {
   const dispatch = useDispatch()
 
+  const registerListeners = useCallback(() => {
+    if (timer) {
+      registerStopListeners()
+    }
+    timer = setTimeout(() => {
+      dispatch(resetDashboardAutoRefresh(dashboard))
+      registerStopListeners()
+    }, autoRefresh.inactivityTimeout)
+
+    window.addEventListener('load', registerListeners)
+    document.addEventListener('mousemove', registerListeners)
+    document.addEventListener('keypress', registerListeners)
+  }, [dashboard, autoRefresh.inactivityTimeout])
+
+  const visChangeHandler = () => {
+    if (document.visibilityState === 'hidden') {
+      registerListeners()
+    } else {
+      registerStopListeners()
+    }
+  }
+  const registerStopListeners = useCallback(() => {
+    // Stop all existing timers and deregister everythang
+    if (timer) {
+      clearTimeout(timer)
+      timer = null
+    }
+
+    window.removeEventListener('load', registerListeners)
+    document.removeEventListener('mousemove', registerListeners)
+    document.removeEventListener('keypress', registerListeners)
+  }, [registerListeners])
+
+  document.addEventListener('visibilitychange', visChangeHandler)
+
+  useEffect(() => {
+    if (
+      autoRefresh?.status &&
+      autoRefresh.status === AutoRefreshStatus.Active &&
+      autoRefresh.inactivityTimeout > 0
+    ) {
+      registerListeners()
+    } else {
+      registerStopListeners()
+    }
+    return () => {
+      registerStopListeners()
+      document.removeEventListener('visibilitychange', visChangeHandler)
+    }
+  }, [autoRefresh?.status, autoRefresh.inactivityTimeout])
+
+  const stopFunc = useCallback(() => {
+    if (autoRefresh?.duration?.upper <= new Date().toISOString()) {
```

Comment:
I could be wrong but I think this will always be false since the comparison is between a string? Might need a ‘getTime’ tacked to it?

Reply:
I was wrong. I didn't realize that that `upper` was being set to ISOString as well. Having said that I think we should probably normalize these times to numbers for the sake of safety if the implementation is iterated upon at a later date and the same thoroughness isn't maintained so that we don't end up in a situation where someone uses a `Date.toString()` and messes everything up.

Reply:
Yup! Addressed, we are going to use getTime() now, so its much more "accurate." This was a hacky solution, thank you for pointing that out!